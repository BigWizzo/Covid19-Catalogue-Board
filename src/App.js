import { Route, Switch } from 'react-router-dom';
import CountryList from './components/CountryList';
import CountryDetails from './components/CountryDetails';

function App() {
  return (
    <div className="App container text-light">
      <Switch>
        <Route path="/" component={CountryList} exact />
        <Route path="/:country" component={CountryDetails} />
      </Switch>
    </div>
  );
}

export default App;
