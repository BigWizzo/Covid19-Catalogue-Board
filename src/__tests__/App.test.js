import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import App from '../App';
import { Provider } from 'react-redux';
import { store, persistor } from '../redux/store';
import { PersistGate } from 'redux-persist/integration/react';

test('should render the Covid19-Catalogue-Board Application', () => {
  const component = renderer.create(
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </PersistGate>
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
