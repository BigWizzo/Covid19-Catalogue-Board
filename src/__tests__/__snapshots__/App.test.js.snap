// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Covid19-Catalogue-Board Application 1`] = `
<div
  className="App container text-light"
>
  <div
    className="main"
  >
    <h1
      className="py-4 text-center"
    >
      Covid19 Catalogue Board 1
    </h1>
    <form
      className="country-form m-auto mb-5"
    >
      <div
        className=""
      >
        <input
          className="form-control form-outline"
          onChange={[Function]}
          placeholder="Search by Country"
          type="text"
          value=""
        />
        <select
          aria-label="Default select example"
          className="form-select p-2"
          onChange={[Function]}
          value=""
        >
          <option
            value="Search by Region"
          >
            Search by Region
          </option>
          <option
            value="Africa"
          >
            Africa
          </option>
          <option
            value="Europe"
          >
            Europe
          </option>
          <option
            value="Asia"
          >
            Asia
          </option>
          <option
            value="North America"
          >
            North America
          </option>
          <option
            value="South America"
          >
            South America
          </option>
          <option
            value="Oceania"
          >
            Oceania
          </option>
        </select>
      </div>
    </form>
    <div
      className="content"
    >
      <div
        className="row g-0"
      />
    </div>
    <div
      className="circle-small-4 br-50 pos-abs"
    />
    <div
      className="circle-small-5 br-50 pos-abs"
    />
    <div
      className="circle-small-2 br-50 pos-abs"
    />
  </div>
</div>
`;
